dashboard-ui/src/components/App/index.jsx

```diff
@@ -144,134 +155,168 @@ export class App extends React.Component {
-                                            </NavDropdown.Item>
-                                        </NavDropdown>
-                                        <NavDropdown title="ADM Evaluation Segments">
-                                            <NavDropdown.Item as={Link} className="dropdown-item" to="/results">
-                                                ADM Data
-                                            </NavDropdown.Item>
-                                            <NavDropdown.Item as={Link} className="dropdown-item" to="/adm-results">
-                                                ADM Alignment Results
-                                            </NavDropdown.Item>
-                                        </NavDropdown>
-                                    </>
-                                )}
-                            </ul>
-                            <ul className="navbar-nav ml-auto">
-                                <li className="login-user">
-                                    <div className="login-user-content">
-                                        <img className="nav-login-icon" src={userImage} alt="" />
-                                        <NavDropdown
-                                            title={currentUser.emails[0].address}
-                                            id="basic-nav-dropdown"
-                                            show={this.state.menuIsOpened}
-                                            onToggle={this.handleToggle}
-                                        >
-                                            <Link className="dropdown-item" to="/myaccount" onClick={this.handleToggle}>
-                                                My Account
-                                            </Link>
-                                            {this.state.currentUser.admin === true && (
-                                                <Link className="dropdown-item" to="/admin" onClick={this.handleToggle}>
-                                                    Administrator
-                                                </Link>
+                <Query query={GET_SURVEY_CONFIG} fetchPolicy={'no-cache'}>
+                    {
+                        ({ loading, error, data }) => {
+                            if (loading) {
+                                return;
+                            }
+                            if (error) {
+                                console.warn("Error getting survey config: ", error);
+                                return;
+                            }
+                            // Get the survey configs one by one, since they are all needed for the results pages
+                            // and add the correct image urls back to them
+                            // then store them properly in localStorage
+                            for (const config of data.getAllSurveyConfigs) {
```

Comment:
This is a preference thing so feel free to ignore. But I'd probably move these for four loops into a separate function like this:
```
function matchImagesToConfig(data) {
    data.getAllSurveyConfigs.forEach(config => {
        config.survey.pages.forEach(page => {
            page.elements.forEach(el => {
                if (el.patients) {
                    el.patients.forEach(patient => {
                        const foundImg = data.getAllImageUrls.find(x => x._id === patient.imgUrl);
                        if (isDefined(foundImg)) {
                            patient.imgUrl = foundImg.url;
                        }
                    });
                }
            });
        });
        store.dispatch(addConfig({ id: config._id, data: config }));
    });
}
```
and then just call it from inside the query tag

Reply:
great call, this was all a result of experimentation and I never really considered how to make it cleaner. Done, but I changed the function name to setupConfigWithImages